### PR TITLE
dash/columnAssignment-datagrid

### PR DESCRIPTION
**Pull request by karolkolodziej**
Added the `visibleColumns` option which allows filtering the columns that should be visible in the `DataGridComponent`.

### CODE REVIEW

**Review by pawellysy — CHANGES_REQUESTED** [review]
I wonder, if this option should be named the same as in the HighchartsComponent. 
In the HighchartsComponent, this option transforms each of the columns to one of the values in the chart, so there are many options like `x`, `y`, `value`, `null` etc, and here there is only one option, `null`. So maybe we should create something like `ignoredColumns: Array<string> where all ignored columns are listed` or `visibleColumns` where the is other way around, or just delete this option from the API and just go with `columns.show = false`? 

Summoning @pawelfus

**Comment by pawelfus** [comment]
Discussion summary: Green light for a different option, instead of `columnAssignment` one

**Review by pawellysy — APPROVED** [review]
LGTM!

File: ts/Dashboards/Plugins/DataGridComponent.ts [hunk 1]
```diff
@@ -378,6 +381,39 @@ class DataGridComponent extends Component {
...
+     *
+     * @internal
+     */
+    private filterColumns(): DataTable|undefined {
+        const table = this.connector?.table.modified,
+            columnsToShow = this.options.columnsToShow;
+
+        if (table) {
+            const columnsToDelete = table.getColumnNames().filter((columnName): boolean => {
+                if (columnsToShow?.length) {
```
Comment (goransle) [inline ts/Dashboards/Plugins/DataGridComponent.ts:395]: The method can probably just return `table` earlier `if( !columnsToShow.length )`.

**Review by bre1470 — COMMENTED** [review]
🙂

**Review by bre1470 — APPROVED** [review]

File: samples/dashboards/cypress/component-datagrid/demo.mjs [hunk 2]
```diff
@@ -82,7 +83,8 @@ Dashboards.board('container', {
             sync: {
                 highlight: true,
                 extremes: true
-            }
+            },
+            columnsToShow: ['Food', 'Vitamin A']
```
Comment (sebastianbochan) [inline samples/dashboards/cypress/component-datagrid/demo.mjs:87]: IMO `visibleColumns` or `displayColumns` sound more intuitive.

**Review by sebastianbochan — APPROVED** [review]

File: samples/dashboards/cypress/component-datagrid/demo.mjs (hunk 2 above)
Comment (sebastianbochan) [reply samples/dashboards/cypress/component-datagrid/demo.mjs:87]: Similar suggestion to `notVisible` param, IMO `hidden` sounds more user-friendly.

**Review by goransle — APPROVED** [review]

**Pull request merged**